### PR TITLE
Handle non-empty Path and empty method path

### LINKED ISSUE #525: Trailing slashes added from registered routes
Hi!

```java
@Path("/webapp/api/p/releasenotes")
public class ReleaseNotesNgController {
    
    @PUT({ "", "/{id: \\d+}" })
    public void save(@Body ReleaseNotesFormDTO dto) {
        //....
    }

}
```

The routes registered for the above example are:
- Add route for PUT '/webapp/api/p/releasenotes/' **(with trailing slash)**
- Add route for PUT '/webapp/api/p/releasenotes/{id: \d+}'

But I believe they should be:
- Add route for PUT '/webapp/api/p/releasenotes' **(no trailing slash)**
- Add route for PUT '/webapp/api/p/releasenotes/{id: \d+}'

What do you think?

### PR DESCRIPTION
**Pull request by munendrasn**
Fixes #525 

We faced this issue during upgrading from 1.7.0 to the latest Pippo version.

### CODE REVIEW

File: pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java [hunk 1]
```diff
@@ -195,11 +192,31 @@ private void registerControllerMethods(Map<Method, Class<? extends Annotation>>
                 // add an empty string to allow controllerPaths iteration
                 controllerPaths.add("");
             }
+            if (methodPaths.length == 0) {
```
Comment (decebals) [inline pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java:195]: I see the same construction for `controllerPath`.  Is it not possible to remove these blocks and to make the code more readable?   For example `if (controllerPaths is empty) { // no controlers => return }`. The same for `methodPaths`.

**Comment by decebals** [comment]
@munendrasn I am happy to see that you added some unit tests :+1:.

File: pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java (hunk 1 above)
Comment (munendrasn) [reply pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java:195]: I could make this change but complexity would remain same. As user could define `@Path("")` or `@GET("")` in which case, controllerPath or methodPath would be non-empty.

I could covert the above using ternary operator
```java
controllerPaths = controllerPaths.isEmpty? Collections.singletonSet(""): controllerPaths
methodPaths = methodPaths.length == 0? new String[]{""}: methodPaths
```
does this look good or I misunderstood?

**Comment by decebals** [comment]
@munendrasn I think that it's ready for merge if @mhagnumdw doesn't see other problems.
@munendrasn Because now you are a committer (congratulations :1st_place_medal:) please merge it yourself (always use `Squash and merge` option because we are not interested in keeping the intermediate committees)

**Comment by munendrasn** [comment]
I will wait for @mhagnumdw's feedback and then merge it

**Comment by mhagnumdw** [comment]
@munendrasn , go ahead.

**Pull request merged**